pyartcd/pyartcd/pipelines/operator_sdk_sync.py

```diff
@@ -0,0 +1,130 @@
+        self._github_client = runtime.new_github_client()
+        self.session = koji.ClientSession("https://brewhub.engineering.redhat.com/brewhub")
+        self.ocp_build_data_repo = self._github_client.get_repo("openshift/ocp-build-data")
+        self.release_file = self.ocp_build_data_repo.get_contents("releases.yml",ref=group)
+        self.release_yaml = yaml.load(self.release_file.decoded_content,Loader=yaml.FullLoader)
+        self.extra_ad_id = self.release_yaml['releases'][assembly]['assembly']['group']['advisories']['extras']
+        self.parent_jira_key = self.release_yaml['releases'][assembly]['assembly']['group']['release_jira']
+        self.imagePath = 'registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-operator-sdk'
+
+
+    def run(self):
+        advisory = Erratum(errata_id = self.extra_ad_id)
+        _LOGGER.info("Check advisory status ...")
+        if advisory.errata_state in ["QE", "NEW_FILES"]:
+            _LOGGER.info("Advisory status not in REL_PREP yet ...")
+            return
+        elif advisory.errata_state == "SHIPPED_LIVE":
+            _LOGGER.info("Advisory status already in SHIPPED_LIVE, update subtask 9 ...")
+            self._update_jira(self.parent_jira_key, 8, "Advisory status already in SHIPPED_LIVE")
+        _LOGGER.info("Advisory status already in post REL_PREP, update subtask 7 ...")
+        self._update_jira(self.parent_jira_key, 6, "Advisory status already in REL_PREP")
+
+        # check if build exist
+        et_builds = advisory.errata_builds
+        sdk_build = [ i for i in et_builds[f'OSE-{self.version}-RHEL-8'] if re.search("openshift-enterprise-operator-sdk-container*", i)]
+        if sdk_build == []:
+            _LOGGER.info("Operator SDK build dosen't attached on extra advisory, subtask 8 then close ...")
+            self._update_jira(self.parent_jira_key, 7, "There isn't a new operator_sdk build in this release's extra advisory. \noperator_sdk_sync job : {}".format(os.environ.get("BUILD_URL")))
+            return
+
+        # get build digest list
+        build = self.session.getBuild(sdk_build[0])
+        archlist = ["amd64", "arm64", "ppc64le", "s390x"]
+        sdkVersion = self._get_sdkversion(build['extra']['image']['index']['pull'][0])
+        _LOGGER.info(sdkVersion)
+        for arch in archlist:
+            self._extract_binaries(arch, sdkVersion, build['extra']['image']['index']['pull'][0])
+        self._update_jira(self.parent_jira_key, 7, "operator_sdk_sync job : {}".format(os.environ.get("BUILD_URL")))
+
+    def _get_sdkversion(self, build):
+        output = subprocess.getoutput(f"oc image info --filter-by-os amd64 -o json {build} | jq .digest")
+        shasum = re.findall("sha256:\w*", output)[0]
+        cmd = f"oc image extract {self.imagePath}@{shasum} --path /usr/local/bin/{self.sdk}:. --confirm && chmod +x {self.sdk} && ./{self.sdk} version && rm {self.sdk}"
+        _LOGGER.info(cmd)
+        sdkvalue = subprocess.getoutput(cmd)
```

Comment:
cmd_gather_async don't support pipeline operation, things that can be solved by one line of code will become more cumbersome

Reply:
@Ximinhan you can try `shell=true` option in `cmd_gather_async` in case if you want to change to that.

Reply:
In the following cmd, I will need to execute multiple shell commands and connect the input and output. If the async method is used, it will split the command and combine it with the communicate() method. I still think the current method is more straightforward for this job.

Reply:
exectools in pyartcd passes almost all parameters to the underlying asyncio.create_subprocess_exec function. It may not have a shell parameter.
We can always extend those functions to call asyncio.create_subprocess_shell if we want the async functionality.